pkg/validation/internal/multiarch.go

```diff
@@ -0,0 +1,513 @@
+			for _, c := range v.Spec.Template.Spec.Containers {
+				data.Images[c.Image] = append(data.Images[c.Image], Platform{})
+			}
+		}
+	}
+
+	for _, v := range data.Bundle.CSV.Spec.RelatedImages {
+		data.Images[v.Image] = append(data.Images[v.Image], Platform{})
+	}
+}
+
+// runManifestInspect executes the command for we are able to check what
+// are the Architecture(s) and SO(s) supported per each image found
+func runManifestInspect(image, tool string) (ManifestInspect, error) {
+	cmd := exec.Command(tool, "manifest", "inspect", image)
+	output, err := runCommand(cmd)
+	if err != nil || len(output) < 1 {
+		return ManifestInspect{}, err
+	}
+
+	var inspect ManifestInspect
+	if err := json.Unmarshal(output, &inspect); err != nil {
+		return ManifestInspect{}, err
+	}
+	return inspect, nil
+}
+
+// run executes the provided command within this context
+func runCommand(cmd *exec.Cmd) ([]byte, error) {
+	command := strings.Join(cmd.Args, " ")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return output, fmt.Errorf("%s failed with error: (%v) %s", command, err, string(output))
+	}
+	return output, nil
+}
+
+// inspectAllImages will perform the required steps to inspect all images found
+func (data *MultiArchValidator) inspectAllImages() {
+	for k, _ := range data.Images {
+		manifest, err := runManifestInspect(k, data.ContainerTool)
+		if err != nil {
+			data.Warns = append(data.Warns, fmt.Errorf("unable to inspect the image (%s) : %s", k, err))
+
+			// We set the Arch and SO as error for we are able to deal witth these cases further
+			// Se that make no sense we raise a warning to notify the user that the image
+			// does not provide some kind of support only because we were unable to inspect it.
+			// Be aware that the validator raise warnings for all cases scenarios to let
+			// the author knows that those were not checked at all and why.
+			data.Images[k] = append(data.Images[k], Platform{"error", "error"})
+			continue
+
+		}
+
+		if manifest.ManifestData != nil {
+			for _, manifest := range manifest.ManifestData {
+				data.Images[k] = append(data.Images[k], manifest.Platform)
+			}
+		}
+	}
+
+}
+
+// checks centralize all checks which are done with this validator
+func (data *MultiArchValidator) checks() {
+	data.checkMissingLabelsForArchs()
+	data.checkMissingLabelsForSO()
+	data.checkMissingImagesSupport()
+	data.checkSupportDefined()
+}
+
+// checkMissingImagesSupport checks if any image is missing some arch or so found
+// (probably the image should support the arch or SO )
+func (data *MultiArchValidator) checkMissingImagesSupport() {
+	for image, arc := range data.Images {
+		listArchNotFound := []string{}
+		for archFromList := range data.AllArchFound {
+			found := false
+			for _, imageData := range arc {
+				// Ignore the case when the Plataform.Architecture == "error" since that means
+				// that was not possible to inspect the image
+				if imageData.Architecture == "error" {
+					found = true
```

Comment:
What do we gain by not breaking here? The only thing that can happen is `found` gets set to `true` a second time.